### DISCUSSION TITLE
This is phenomenal! Questions

### DISCUSSION THREAD

**Discussion by twardoch** [neonbjb/tortoise-tts#77] [General]
Your project is a breakthrough. I’ve recorded myself reading the same short 4 paragraphs twice, with different prosodies, and split it into some 20 clips, each about 10 seconds. 

I then edited the notebook code to use those clips as reference. I did two variants: one where I made just once voice with 20 clips, and one where I made 2 voices, with 10 clips each (split by the prosody). 

I must say the 2nd variant works much better! 

I have two questions: 

1. What’s a reasonable length of text to process in one go? 
2. I split a longer texts into short paragraphs with 1-3 sentences each, and processed each paragraph separately. Weirdly enough, in one of the resulting clips, the result switches between sentences from my voice (in fact the mix of two of my voices) into a female voice. What can be the reason for it? 

Overall, the results are spectacular. I can say that I nearly can’t distinguish the synthesized clips from my own real voice, and I’ve recorded quite a lot myself and know what I sound like :) 

This is the first opwnsource TTS tool that’s over heard that truly delivers »human sound«. And its cloning quality is incredible. 

Many many thanks for your hard work on this!

**Comment by twardoch** [comment]
Ps. I’ll publish my samples & code in the next days as reference. 

And I’ll add: for the “cloning”, I first used 3 short samples of my recording, and the result didn’t sound very much like me (but a bit). Yet with the 2x10 clips approach split into 2 “voices”, it’s much better!

**Comment by neonbjb** [comment]
Thanks for the compliments!

Regarding voice switching, others have noted that this appears to happen more often after a period, so one way you can get around it is by breaking up text into sentences. Honestly the only other way is to manually prune these results out at the moment. I believe this is primarily a defect attributed to my training dataset.

(I'm converting this to a "discussion" since there's nothing I can fix here, hope you don't mind)

**Reply by twardoch** [reply]
I'll do some more functional testing. Some of the phonemes are not mine in my first test, so I'll give it a recording of some English phonetic pangrams like https://clagnut.com/blog/2380/#English_phonetic_pangrams — they're short but are supposed to cover the whole spectrum. I wonder if this will improve the result. 😁

**Reply by vinithkac** [reply]
Did it improve results?
